### PR TITLE
feat(modelarts): add a new resource to manage network

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccModelartsNetwork_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccModelartsNetwork_basic
=== PAUSE TestAccModelartsNetwork_basic
=== CONT  TestAccModelartsNetwork_basic
--- PASS: TestAccModelartsNetwork_basic (310.37s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	310.418s

```

### CODE REVIEW

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_network.go [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+	getModelartsNetworkResp, err := getModelartsNetworkClient.Request("GET", getModelartsNetworkPath, &getModelartsNetworkOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Modelarts network")
+	}
+
+	getModelartsNetworkRespBody, err := utils.FlattenResponse(getModelartsNetworkResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```
Comment (houpeng80) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_network.go]: how about extracting these codes into a separate function, the other three waiting functions all use these codes

Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_network.go:291]: OK

Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_network.go]: ok

**Pull request merged**